### PR TITLE
HPCC-15062 Design and implement parallel child queries

### PR DESCRIPTION
**Pull request by richardkchapman**
HPCC-15062 Design and implement parallel child queries

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: roxie/ccd/ccdserver.cpp [hunk 1]
```diff
@@ -1663,7 +1660,10 @@ class CRoxieServerStrandedActivity : public CRoxieServerActivity
         active = 0;
     }
 
-    //This function is pure (But also implemented out of line) to force the derived classes to implement it.
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp]: Was this comment line deliberately removed?

File: roxie/ccd/ccdserver.cpp [hunk 2]
```diff
@@ -26735,6 +26735,64 @@ class CActivityGraph : public CInterface, implements IActivityGraph, implements
...
+    virtual IEclGraphResults * evaluate(unsigned parentExtractSize, const byte * parentExtract)
+    {
+        Owned<CActivityGraph> realGraph;
+        {
+            CriticalBlock b(graphCrit);
+            if (stack.length())
+                realGraph.setown(&stack.popGet());
+            else
+            {
+                realGraph.setown(new CActivityGraph(ctx, graphName, id, graphDefinition, NULL, logctx));
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp]: I think this could be outside the critical section.  That may help start up costs marginally.

**Comment by ghalliday** [comment]
@richardkchapman a couple of minor questions/comments

File: roxie/ccd/ccdserver.cpp (hunk 1 above)
Comment (richardkchapman) [inline roxie/ccd/ccdserver.cpp]: Looks like it has become separated from the function it was supposed to refer to... I will fix

**Comment by richardkchapman** [comment]
@ghalliday Repushed to address comments

**Pull request merged**